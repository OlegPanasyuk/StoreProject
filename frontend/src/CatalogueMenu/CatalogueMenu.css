:root {
    --storesecondary: #385c80;
}

.bg-storesecondary {
    background: var(--storesecondary);
}

.catalogue__list-item {
    cursor: pointer
}

.catalogue__list-item:hover {
    background: var(--storesecondary);
} 