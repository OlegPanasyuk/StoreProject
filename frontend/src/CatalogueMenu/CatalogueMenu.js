import React from 'react';
import { ListGroup } from 'react-bootstrap';
import PropTypes from 'prop-types';


function CatalogueMenu(props) {
    const obj = props.obj;
    const handleClick = props.handleClick;
    const requestGoods = props.requestGoods;
    
    let arrItems = Object.keys(obj).map((el) => {
        let element;
        if (obj[el].children && (Object.keys(obj[el].children).length > 0)) {
            element = (
                <ListGroup.Item 
                    onClick={
                        ()=>handleClick(el)
                    } 
                    variant="secondary" 
                    key={el}>
                    {`${obj[el].id_catalogue}  ${obj[el].name}`}
                </ListGroup.Item>
                
            );    
        } else {
            element = (
                <ListGroup.Item 
                    onClick={
                        ()=> {
                            handleClick(el);
                            requestGoods(el);
                        }
                    } 
                    variant="secondary" 
                    key={el}>
                    {`${obj[el].id_catalogue}  ${obj[el].name}`}
                </ListGroup.Item>
            );  
        }
        
        return element;
    });
    return (
        <React.Fragment>
            <ListGroup>
                {arrItems}
            </ListGroup>
        </React.Fragment>
    );
}

CatalogueMenu.propTypes = {
    obj : PropTypes.object
};

export default CatalogueMenu;