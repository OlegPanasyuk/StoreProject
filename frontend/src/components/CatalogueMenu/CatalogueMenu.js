import React from 'react';
import { ListGroup } from 'react-bootstrap';
import PropTypes from 'prop-types';
import './CatalogueMenu.css';


function CatalogueMenu({
    obj,
    handleClick,
    requestGoods
}) {
    const arrItems = Object.keys(obj).map((el) => {
        let element;
        if (obj[el].children && (Object.keys(obj[el].children).length > 0)) {
            element = (
                <ListGroup.Item
                    className='catalogue__list-item'
                    onClick={() => {
                        handleClick(el);
                        requestGoods(el);
                    }}

                    key={el}
                >
                    {`${obj[el].name}`}
                </ListGroup.Item>

            );
        } else {
            element = (
                <ListGroup.Item
                    className='catalogue__list-item'
                    onClick={() => {
                        handleClick(el);
                        requestGoods(el);
                    }}

                    key={el}
                >
                    {`${obj[el].name}`}
                </ListGroup.Item>
            );
        }

        return element;
    });
    return (
        <React.Fragment>
            <ListGroup>
                {arrItems}
            </ListGroup>
        </React.Fragment>
    );
}

CatalogueMenu.propTypes = {
    obj: PropTypes.object,
    handleClick: PropTypes.func,
    requestGoods: PropTypes.func
};

CatalogueMenu.defaultProps = {
    obj: {},
    handleClick: () => {},
    requestGoods: () => {}
};

export default CatalogueMenu;
