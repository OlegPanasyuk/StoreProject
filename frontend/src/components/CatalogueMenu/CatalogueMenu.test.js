import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import CatalogueMenu from './CatalogueMenu';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    obj: {
        1: {
            id_catalogue: 1,
            name: 'Car',
            description: 'Group 1',
            parent_id: -1
        },
        2: {
            id_catalogue: 2,
            name: 'BigCars',
            description: 'Group 2',
            parent_id: -1
        }
    },
    handleClick: (i) => { alert(`Click by Item Menu${i}`); },
    requestGoods: () => {}
}) {
    const component = shallow(<CatalogueMenu {...props} />);
    return { component, props };
}

describe('CatalogueMenu', () => {
    it('Component must be created and props must be correct', () => {
        const { component } = setupComponent();
        expect(CatalogueMenu.propTypes.obj).toBe(PropTypes.object);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});
