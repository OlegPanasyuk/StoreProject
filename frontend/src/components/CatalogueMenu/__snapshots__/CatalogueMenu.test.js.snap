// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogueMenu Component must be created and props must be correct 1`] = `
<Fragment>
  <ForwardRef>
    <Bootstrap(ListGroupItem)
      className="catalogue__list-item"
      key="1"
      onClick={[Function]}
    >
      Car
    </Bootstrap(ListGroupItem)>
    <Bootstrap(ListGroupItem)
      className="catalogue__list-item"
      key="2"
      onClick={[Function]}
    >
      BigCars
    </Bootstrap(ListGroupItem)>
  </ForwardRef>
</Fragment>
`;
